roles/common/tasks/main.yml

```diff
@@ -3,24 +3,38 @@
   fail: msg='This role only supports Yum or Apt package managers at the moment.  Please do a manual install instead.'
   when: ansible_pkg_mgr != 'yum' and ansible_pkg_mgr != 'apt'
 
-- name: yum install Java 1.7 and some basic dependencies
+- name: Fetch Java version
+  shell: java -version 2>&1 | grep version | awk '{print $3}' | sed 's/"//g' | grep '^.\..' -o || echo 0.0
```

Comment:
okay, cool, thanks for those tips. i'm no longer actively using this playbook, so i can't test it, but if you are confident of course feel free to adapt and merge.